### PR TITLE
Show error in case of failing to load a shared dashboard

### PR DESCRIPTION
**Pull request by arikfr**
## What type of PR is this? (check all applicable)

- [x] Other

## Mobile & Desktop Screenshots/Recordings (if there are UI changes)

image

### CODE REVIEW

File: client/app/pages/dashboards/public-dashboard-page.js [hunk 1]
```diff
@@ -52,6 +53,8 @@ const PublicDashboardPage = {
         };
 
         this.extractGlobalParameters();
+      }).catch((error) => {
+        throw new PromiseRejectionError(error);
```
Comment (ranbena) [inline client/app/pages/dashboards/public-dashboard-page.js]: Do we have deliberate relevant exceptions where we could insert meaningful error messages to?

Comment (arikfr) [reply client/app/pages/dashboards/public-dashboard-page.js]: `PromiseRejectionError` is actually such an exception: https://github.com/getredash/redash/blob/master/client/app/lib/promise-rejection-error.js

Comment (ranbena) [reply client/app/pages/dashboards/public-dashboard-page.js]: I mean the `error` we catch - make it's message be meaningful.
"...page cannot be found" is just confusing in this case.

Comment (arikfr) [reply client/app/pages/dashboards/public-dashboard-page.js]: What would you expect here?

Comment (ranbena) [reply client/app/pages/dashboards/public-dashboard-page.js]: Oh I see there's a relevant error message a few lines above:
https://github.com/getredash/redash/blob/2ca763747a787d8a9af90189953440745fcfa238/client/app/pages/dashboards/public-dashboard-page.js#L45

 BUT it seems to be doing it wrong, as PromiseRejectionError requires
1. `rejection.status !== undefined/404/403`
2. `rejection.data.error = 'the message'`

Comment (arikfr) [reply client/app/pages/dashboards/public-dashboard-page.js]: But this is not what this change about. This change about showing a 404 when you use a bad token (either non existing or just invalid).

Comment (ranbena) [reply client/app/pages/dashboards/public-dashboard-page.js]: Ok 😭

**Pull request merged**